src/main/java/it/gov/pagopa/admissibility/rest/anpr/residence/ResidenceAssessmentRestClientImpl.java

```diff
@@ -83,14 +93,18 @@ public Mono<RispostaE002OKDTO> getResidenceAssessment(String accessToken, String
                     throw new AnprDailyRequestLimitException(e);
                 });
         //TODO define error code for retry
+    }
 
+    private Mono<RichiestaE002DTO> generateRequest(String fiscalCode) {
+        return customSequenceGeneratorRepository.generateSequence(OnboardingConstants.GLOBAL_SEQUENCE_ID)
+                .map(sequenceValue -> createRequest(fiscalCode, sequenceValue));
     }
 
-    private RichiestaE002DTO generateRequest(String fiscalCode) {
+    private RichiestaE002DTO createRequest(String fiscalCode, Long sequenceValue) {
         Date dateNow = new Date();
-        String dateWithHoursString = new  SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss").format(dateNow);
+        String dateWithHoursString = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss").format(dateNow);
```

Comment:
let's store new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss" into a const variable of this class